pkg/volume/glusterfs/glusterfs.go

```diff
@@ -436,6 +439,89 @@ func (d *glusterfsVolumeDeleter) GetPath() string {
 	return d.plugin.host.GetPodVolumeDir(d.glusterfsMounter.glusterfs.pod.UID, strings.EscapeQualifiedNameForDisk(name), d.glusterfsMounter.glusterfs.volName)
 }
 
+//
+// Traverse the PVs, fetching all the GIDs from those
+// in a given storage class, and mark them in the table.
+//
+func (p *glusterfsPlugin) collectGids(class *storage.StorageClass) error {
+	pvList, err := p.host.GetKubeClient().Core().PersistentVolumes().List(v1.ListOptions{LabelSelector: labels.Everything().String()})
+	if err != nil {
+		glog.Errorf("glusterfs: failed to get existing persistent volumes")
+		return err
+	}
+
+	var gidStrings []string
+	gidStrings = []
+
+	for _, pv := range pvList.Items {
+		if pv.Annotations[annDynamicallyProvisioned] != glusterfsPluginName {
```

Comment:
I don't think we care about this... if a PV is tied to the storage class and has a gid, we want to add it